backend/src/v2/driver/k8s.go

```diff
@@ -482,6 +482,87 @@ func extendPodSpecPatch(
 		podSpec.Containers[0].VolumeMounts = append(podSpec.Containers[0].VolumeMounts, emptyDirVolumeMount)
 	}
 
+	// Get node affinity information
+	if nodeAffinityTerms := kubernetesExecutorConfig.GetNodeAffinity(); len(nodeAffinityTerms) > 0 {
+		var requiredTerms []k8score.NodeSelectorTerm
+		var preferredTerms []k8score.PreferredSchedulingTerm
+
+		for _, nodeAffinityTerm := range nodeAffinityTerms {
```

Comment:
It might be worth adding a quick check at the start of the loop to skip empty terms early on:

```go
for i, nodeAffinityTerm := range nodeAffinityTerms {
    if nodeAffinityTerm.GetNodeAffinityJson() == nil && 
       len(nodeAffinityTerm.GetMatchExpressions()) == 0 && 
       len(nodeAffinityTerm.GetMatchFields()) == 0 {
        glog.Warningf("NodeAffinityTerm %d is empty, skipping", i)
        continue
    }
    // rest of processing...
}
```

Right now, we only catch empty explicit terms later in the function, but JSON terms could also be empty and we'd still try to process them. This would save some unnecessary work and give clearer debugging info.